### PR TITLE
chore(docker): use ubuntu:24.04 in forest dockerfiles

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

Changes introduced in this pull request:

-

## Reference issue to close (if applicable)

Closes https://github.com/ChainSafe/forest/issues/4730

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by LesnyRumcajs — APPROVED** [review]

**Comment by LesnyRumcajs** [comment]
No green checkmark!

File: Dockerfile [hunk 1]
```diff
@@ -20,15 +20,16 @@
 
 # Cross-compilation helpers
 # https://github.com/tonistiigi/xx
-FROM --platform=$BUILDPLATFORM ghcr.io/lesnyrumcajs/xx:1.2.1 AS xx
+FROM --platform=$BUILDPLATFORM tonistiigi/xx:1.5.0 AS xx
```
Comment (hanabi1224) [inline Dockerfile:23]: @LesnyRumcajs Is it OK to use upstream xx?

File: Dockerfile [hunk 2]
```diff
@@ -20,15 +20,16 @@
...
 
-FROM --platform=$BUILDPLATFORM ubuntu:22.04 AS build-env
+FROM --platform=$BUILDPLATFORM ubuntu:24.04 AS build-env
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
 # install dependencies
 RUN apt-get update && \
-    apt-get install --no-install-recommends -y build-essential clang curl git ca-certificates
+    apt-get install --no-install-recommends -y build-essential clang-14 curl git ca-certificates
```
Comment (hanabi1224) [inline Dockerfile:30]: clang in ubuntu:24.04 compiles jemalloc with errors, use clang-14 instead

File: Dockerfile (hunk 1 above)
Comment (hanabi1224) [reply Dockerfile]: reverted.

**Review by LesnyRumcajs — APPROVED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -22,13 +22,14 @@
...
 RUN update-ca-certificates
+RUN ln -sf /usr/bin/clang-14 /usr/bin/clang && ln -sf /usr/bin/clang-14++ /usr/bin/clang++
```
Comment (aatifsyed) [inline Dockerfile:32]:
```suggestion
RUN ln -sf /usr/bin/clang-14 /usr/local/bin/clang && ln -sf /usr/bin/clang-14++ /usr/local/bin/clang++
```

It's nonstandard

Comment (hanabi1224) [reply Dockerfile]: @aatifsyed Unfortunately this does not work.

https://github.com/ChainSafe/forest/actions/runs/10718952448/job/29721960709?pr=4731#step:9:1200
```
0.057 xx-cargo install --locked --path . --force
0.068 /usr/bin/clang not found
0.069 make: *** [Makefile:4: install-xx] Error 1`
```

**Review by ruseinov — APPROVED** [review]

**Pull request merged**